### PR TITLE
feat(scmpuff): add fish integration flag

### PR DESCRIPTION
**Pull request by cpcloud**
### Description

Add fish integration to the `scmpuff` program

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: tests/modules/programs/scmpuff/fish.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    programs = {
+      scmpuff.enable = true;
+      fish.enable = true;
+    };
+
+    # Needed to avoid error with dummy fish package.
+    xdg.dataFile."fish/home-manager_generated_completions".source =
+      lib.mkForce (builtins.toFile "empty" "");
+
+    test.stubs.fish = { };
```
Comment (berbiche) [inline tests/modules/programs/scmpuff/fish.nix:12]:
```suggestion
    test.stubs.fish = { };
    test.stubs.scmpuff = { };
```
Is there anything prevent the scmpuff package from being overriden?

File: tests/modules/programs/scmpuff/fish.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    nmt.script = ''
+      assertFileExists home-files/.config/fish/config.fish
+      assertFileContains \
+        home-files/.config/fish/config.fish \
+        '${pkgs.scmpuff}/bin/scmpuff init -s --shell=fish | source'
```
Comment (berbiche) [inline tests/modules/programs/scmpuff/fish.nix:18]:
```suggestion
        '@scmpuff@/bin/scmpuff init -s --shell=fish | source'
```
If the package is overriden with `test.stubs`, its `outPath` will be changed to `@scmpuff@` by default.

File: tests/modules/programs/scmpuff/no-fish.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    # Needed to avoid error with dummy fish package.
+    xdg.dataFile."fish/home-manager_generated_completions".source =
+      lib.mkForce (builtins.toFile "empty" "");
+
+    test.stubs.fish = { };
+
+    nmt.script = ''
+      assertFileNotRegex home-files/.config/fish/config.fish '${pkgs.scmpuff}'
+    '';
```
Comment (berbiche) [inline tests/modules/programs/scmpuff/no-fish.nix:19]: Same thing about overriding the scmpuff package

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: tests/modules/programs/scmpuff/fish.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{ pkgs, lib, ... }: {
+  config = {
```
Comment (rycee) [inline tests/modules/programs/scmpuff/fish.nix:2]: Can omit this `config = {`:

```suggestion
```

File: tests/modules/programs/scmpuff/no-fish.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{ pkgs, lib, ... }: {
+  config = {
```
Comment (rycee) [inline tests/modules/programs/scmpuff/no-fish.nix:2]:
```suggestion
```

File: tests/modules/programs/scmpuff/fish.nix (hunk 4 above)
Comment (cpcloud) [reply tests/modules/programs/scmpuff/fish.nix:2]: Done.

File: tests/modules/programs/scmpuff/fish.nix (hunk 1 above)
Comment (cpcloud) [reply tests/modules/programs/scmpuff/fish.nix:12]: Done

File: tests/modules/programs/scmpuff/fish.nix (hunk 2 above)
Comment (cpcloud) [reply tests/modules/programs/scmpuff/fish.nix:18]: Done

File: tests/modules/programs/scmpuff/no-fish.nix (hunk 5 above)
Comment (cpcloud) [reply tests/modules/programs/scmpuff/no-fish.nix:2]: Done

File: tests/modules/programs/scmpuff/no-fish.nix (hunk 3 above)
Comment (cpcloud) [reply tests/modules/programs/scmpuff/no-fish.nix:19]: Done

**Review by sumnerevans — APPROVED** [review]
Looks OK to me

**Comment by rycee** [comment]
Yep, LGTM! Merged to master now. Thanks!

**Pull request merged**